src/createEpicMiddleware.js

```diff
@@ -54,7 +54,22 @@ export function createEpicMiddleware(rootEpic, options = defaultOptions) {
         switchMap(output$ => options.adapter.output(output$))
       );
 
-      result$.subscribe(store.dispatch);
+      if (process.env.NODE_ENV !== 'production') {
+        // Check if actions returned from epics contain a property
+        // that only the actions going to epics should have.
+        // If present this indicates an infinite loop
+        result$.pipe(
+          require('rxjs/internal/operators/filter').filter(action => {
```

Comment:
We shouldn't reference operators via the internal path as it's subject to change without notice. I imagine you did it this way to avoid pulling it in on production builds, but I think it's unavoidable